### PR TITLE
Webhooks are always active

### PR DESCRIPTION
**Pull request by magul**
giphy

### CODE REVIEW

File: pages/api/manifest.ts [hunk 1]
```diff
@@ -8,29 +8,24 @@ import * as GeneratedGraphQL from "../../generated/graphql";
...
   const entries = await fg(['*.ts'], { cwd: 'pages/api/webhooks' });
 
-  const dropFileExtension = (filename: string) => path.parse(filename).name;
-  const r = entries.map(dropFileExtension).map((name: string) => {
-
+  return entries.map(dropFileExtension).map((name: string) => {
     const camelcaseName = name.split('-').map(capitalize).join('');
     const statement = `${camelcaseName}SubscriptionDocument`;
-    const query = statement in  GeneratedGraphQL ? print((GeneratedGraphQL as any)[statement]) : '';
+    const query = statement in  GeneratedGraphQL ? print((GeneratedGraphQL as any)[statement]) : null;
```
Comment (magul) [inline pages/api/manifest.ts:19]: one more comment here @zaiste , it was decided that manifest-based webhooks will require query to be in place, so we should figure out how to gracefully fail, when there's no subscription query in place, but I don't exactly know how we want to do so?

Comment (magul) [reply pages/api/manifest.ts:19]: I mean, with current setup this issue will appear only when requesting manifest, I believe it should be also failing when running `pnpm dev` or `pnpm build`.

File: pages/api/manifest.ts [hunk 2]
```diff
@@ -8,29 +8,24 @@ import * as GeneratedGraphQL from "../../generated/graphql";
...
 
     return {
       name,
       asyncEvents: [name.toUpperCase().replace("-", "_")],
       query, 
       targetUrl: `${baseURL}/api/webhooks/${name}`,
-      isActive: true,
```
Comment (zaiste) [inline pages/api/manifest.ts:27]: @magul are you sure? when removed, testers won't be able to test it nor specify existing handlers as `active`

**Review by zaiste — CHANGES_REQUESTED** [review]

File: pages/api/manifest.ts (hunk 1 above)
Comment (zaiste) [reply pages/api/manifest.ts:19]: this API is still available as a Feature Preview so we don't need to decide right away. It'd be great that `query` equals `null` means a default, **full** payload. This should be implemented in the core. cc @patrys @timuric

File: pages/api/manifest.ts (hunk 2 above)
Comment (magul) [reply pages/api/manifest.ts:27]: That was requested by @patrys, so maybe you will convice him that we still want it here.

File: pages/api/manifest.ts (hunk 1 above)
Comment (magul) [reply pages/api/manifest.ts:19]: So I don't recall who made this decision, that `query` here will be required. If we will make it optional here, we could handle `syncEvents` here also (there was a discussion yesterday, also with @patrys  and yesterday was decided to just handle `asyncEvents` in manifest for the time being).

> It'd be great that query equals null means a default, full payload

I don't know, what does it mean full payload in this context, but before introduction of subscriptions we have (I called them at least as) legacy payloads and if query is not defined, that legacy payload is delivered.
The other question here is default payload for newly introduced eventtypes.

File: pages/api/manifest.ts (hunk 2 above)
Comment (zaiste) [reply pages/api/manifest.ts:27]: Yes, I've been thinking about something similar, i.e.

`export const isActive = true`

Comment (patrys) [reply pages/api/manifest.ts:27]: All webhook handlers should either be active on installation or dropped from the manifest. When reinstalling an app, I suggest that all webhooks but the ones from the manifest are removed from Saleor before triggering the installation event. In other words, after a successful (re)installation, Saleor should have precisely the set of handlers defined in the manifest.

**Review by Cloud11PL — APPROVED** [review]

**Pull request merged**